extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js

```diff
@@ -170,6 +177,27 @@ class LifecycleDeleteObjectTask extends BackbeatTask {
         });
     }
 
+    /**
+     * Throws error if object has a 'PENDING' replication status
+     * @param {ActionQueueEntry} entry entry object
+     * @param {Logger} log logger instance
+     * @param {Function} done callback
+     * @returns {undefined}
+     */
+    _checkReplicationStatus(entry, log, done) {
```

Comment:
So we decided to evaluate the replication status "late" to prevent an extra metadata call. Did anything else prevent us to do it one step earlier in task/LifecycleTask.js?

Reply:
Actually in all cases we only do one metadata call (the value is kept from the first call and reused later), and so here i don't think it really makes a difference wether we check the replication before or after object lock, what do you think ?

Reply:
I meant "late" by one async communication later. Basically, why don't we check it in the Lifecycle Bucket Processor, after consuming the bucket task topic?

Reply:
Yes as you said the only reason is to save the metadata call nothing more

Reply:
Ok, thanks, I added a note for it on my side.

Context: we are implementing new object listings (specific to lifecycle) that, inter alia, return extra object metadata fields to avoid those extra MD calls. So, we will add Replication Status as one of the objects MD fields returned from the list.

Reply:
Got it, thanks for the context